File: ChatBotFront/src/app/Shared/Models/MessageDto.ts
export class MessageDto {
    username: string;
    messageBody: string;
    historyDto: {userName: string, messageBody: string}[]

    constructor(userResponse: any){
        this.username = userResponse.userName;
        this.messageBody = userResponse.messageBody;
        this.historyDto = userResponse.historyDto;
    }
}

File: ChatbotApi/Dto/MessageDto.cs
using ChatbotApi.Models;
using System.Collections.Generic;

namespace ChatbotApi.Dto
{
    public class MessageDto
    {
        public string UserName { get; set; }
        public string MessageBody { get; set; }
        public List<Message> HistoryDto { get; set; }
    }
}
